tmt/cli.py

```diff
@@ -2043,17 +2050,18 @@ def clean_guests(
     if last and bool(id_):
         raise tmt.utils.GeneralError(
             "Options --last and --id cannot be used together.")
-    if not Path(workdir_root).exists():
+    if workdir_root and not Path(workdir_root).exists():
         raise tmt.utils.GeneralError(f"Path '{workdir_root}' doesn't exist.")
 
     assert context.obj.clean_logger is not None  # narrow type
-
+    workdir_root_path = Path(workdir_root) if workdir_root else None
```

Comment:
This expression can really live in `tmt.Clean()` call below, `workdir_root_path` has no other purpose in life than to be passed to `Clean` class. `workdir_root=effective_workdir_root(Path(workdir_root) if workdir_root else None))` is perfectly fine.

Reply:
Ha, nevermind. How about doing it slightly differently? `workdir_root` and `workdir_root_path` - and also `root_path` above, in `root_path = effective_workdir_root(workdir_root_path)`, the naming is a bit fractured. We need to invent new names because `workdir_root` holds `Optional[str]`, but there's `effective_workdir_root()`, `--workdir-root`, `WORKDIR_ROOT`, so suddenly using `root_path` is just puzzling and taking a turn to new namespace.

Buw, we can ask `option()` to use different name, IIRC, doing this: `option('--workdir-root', '_workdir_root', ...)`. Commands accepting this option then need to change their `workdir_root` parameter to `_workdir_root`, and then we can easily convert `_workdir_root` to a `Path` instance and be done with this mess:

```python
...
@workdir_root_options
@verbosity_options
@dry_options
def clean(context: Context,
          last: bool,
          id_: tuple[str, ...],
          skip: list[str],
          _workdir_root: Optional[str],
          **kwargs: Any) -> None:
    ...
    workdir_root = Path(_workdir_root) if _workdir_root is not None else None

    if workdir_root and not workdir_root.exists():
        raise tmt.utils.GeneralError(f"Path '{workdir_root}' doesn't exist.")

    ...

    if context.invoked_subcommand is None:
        assert context.obj.clean_logger is not None  # narrow type
        root_path = effective_workdir_root(workdir_root)
        ...
```

No unexpected names, the conversion happens as early as possible, workdir root is in `workdir_root`, not in `root_dir` or `workdir_root_path`, and the world will be a bit cleaner. WDYT?